### DISCUSSION TITLE
Can I contribute for version 2.x?

### DISCUSSION THREAD

**Discussion by juan-pinzon** [ionic-team/capacitor#6322] [Q&A]
In my project we are still using capacitor version 2, I already plan to upgrade to 3, but not too soon. By Google Play policies I changed the Android target to version 31 and with that you must explicitly place the android:exported (https://developer.android.com/about/versions/12/behavior-changes-12?hl=es-419#exported) so I would like to be able to contribute in the version by placing that property in the capacitor/android/capacitor/src/main/AndroidManifest.xml file in the service with name capacitor/android/capacitor/src/main/AndroidManifest.xml

**Comment by jcesarmobile** [comment]
As you can see on the CONTRIBUTIN.md file, 2.x branch is no longer maintained
https://github.com/ionic-team/capacitor/blob/main/CONTRIBUTING.md#branches

that's not the only change needed for supporting target SDK 31 or 32, you also need to fix share, local-notifications, push-notifications and a few other plugins that require code changes when targeting SDK 31 or 33. Also a few other changes are required if targeting SDK 33 (which will be officially supported in Capacitor 5 and required by Google starting in August for new apps and November for existing apps)
That's why we make yearly major releases, because every year Google and Apple require to use their latest SDKs and specially for Android, those changes are breaking, so that's why the release is a major release. And makes no sense to maintain the old versions once November is reached.

You can still fork and maintain the changes for yourself or share them with others, but we can't make breaking changes in the 2.x branch, and the required changes to support SDK 31, 32 and 33 are breaking.
